trin-storage/src/versioned/id_indexed_v1/store.rs

```diff
@@ -441,26 +455,27 @@ impl IdIndexedV1Store {
         let conn = self.config.sql_connection_pool.get()?;
         let mut delete_query = conn.prepare(&sql::delete_farthest(&self.config.content_type))?;
 
-        while self
-            .usage_stats
-            .is_above(self.config.storage_capacity_bytes)
-        {
-            let to_delete = self
-                .config
-                .estimate_to_delete_until_target(&self.usage_stats)
-                .min(MAX_TO_PRUNE_PER_QUERY);
+        while self.pruning_strategy.should_prune(&self.usage_stats) {
+            let to_delete = self.pruning_strategy.get_pruning_count(&self.usage_stats);
 
             if to_delete == 0 {
-                error!(Db = %self.config.content_type, "Should delete 0. This is not expected to happen (we should be above storage capacity).");
+                error!(
+                    Db = %self.config.content_type,
+                    "Entries to prune is 0. This is not supposed to happen (we should be above storage capacity)."
+                );
                 return Ok(());
             }
 
+            let pruning_start_time = Instant::now();
             let delete_timer = self.metrics.start_process_timer("prune_delete");
-            let params = named_params! { ":limit": to_delete };
             let deleted_content_sizes = delete_query
-                .query_map(params, |row| row.get("content_size"))?
+                .query_map(named_params! { ":limit": to_delete }, |row| {
+                    row.get("content_size")
+                })?
                 .collect::<Result<Vec<u64>, rusqlite::Error>>()?;
             self.metrics.stop_process_timer(delete_timer);
+            self.pruning_strategy
+                .observe_pruning_duration(pruning_start_time.elapsed());
```

Comment:
Seems like a bummer to have prune run its own `Instant` timer, when we're already calculating the time in the metrics. I suppose if `stop_process_timer` returned the duration it observed (looks like it's an f64 in seconds, but we could convert to a `Duration`), then `observe_duration` could reuse it. Do you agree that would be preferable? If you don't feel like doing that, I might do it after this merges, just so it's ready the next time we want to re-use the timing.